### PR TITLE
Untangle eg.makeUser names; simplify some tests

### PR DESCRIPTION
**Pull request by gnprice**
I was looking at the tests in `userHelpers-test.js` while reviewing #5292 (which added a few more tests there), and after merging that I wanted to make that file simpler and better set up to add more such tests.

Along the way I realized that `eg.makeUser` has a hidden gotcha in the `name` parameter, so I made a series of commits to untangle that.

There are still further improvements I might make to this file that I didn't get to today, but I think this is significantly better already.

### CODE REVIEW

File: src/chat/__tests__/narrowsSelectors-test.js [hunk 1]
```diff
@@ -334,159 +334,54 @@ describe('getStreamInNarrow', () => {
...
-      },
-      streams: [],
-      users: [],
-    });
-    const narrow = pm1to1NarrowFromUser(bot);
-
-    const result = isNarrowValid(state, narrow);
-
-    expect(result).toBe(true);
+    const narrow = pm1to1NarrowFromUser(eg.crossRealmBot);
```
Comment (chrisbobbe) [inline src/chat/__tests__/narrowsSelectors-test.js:377]:
```
narrow tests [nfc]: Simplify and compact, using example data
```

Wow, lots of opportunity to be compact here; nice.

File: src/__tests__/lib/exampleData.js [hunk 2]
```diff
@@ -150,18 +150,17 @@ const randUserId: () => UserId = (mk => () => makeUserId(mk()))(
...
     avatar_url: args.avatar_url ?? makeAvatarUrl(user_id.toString()),
 
     date_joined: `2014-04-${randInt(30)
       .toString()
       .padStart(2, '0')}`,
 
-    email: args.email ?? `${name}@example.org`,
-    full_name: args.full_name ?? `${capsName} User`,
+    email: args.email ?? `${randName}@example.org`,
+    full_name: args.full_name ?? `${randName} User`,
```
Comment (chrisbobbe) [inline src/__tests__/lib/exampleData.js:163]:
```
example data [nfc]: Simplify random naming in makeUser
```

Tiny nit: I think this isn't quite NFC, in that when `args.full_name` isn't passed, the `full_name` property will no longer be guaranteed to have an uppercase first letter. (I think this is a good change, I just think it's slightly non-NFC in that way.)

**Review by chrisbobbe — APPROVED** [review]
Great, thanks! Please merge at will.

File: src/__tests__/lib/exampleData.js (hunk 2 above)
Comment (gnprice) [reply src/__tests__/lib/exampleData.js:163]: True. I'll tweak that commit message.

**Comment by gnprice** [comment]
Thanks for the review!

**Pull request merged**